timemachine/ff/handlers/openmm_deserializer.py

```diff
@@ -116,13 +121,17 @@ def deserialize_system(system: mm.System, cutoff: float) -> Tuple[List[potential
                 sig = value(sig)
                 eps = value(eps)
 
-                # increment eps by 1e-3 if we have eps==0 to avoid a singularity in parameter derivatives
-                # override default amber types
-
-                # this doesn't work for water!
-                # if eps == 0:
-                # print("Warning: overriding eps by 1e-3 to avoid a singularity")
-                # eps += 1e-3
+                # override (sig, eps) if we have eps==0, to avoid singularities in potential and derivatives
+                sig_override = 1e-4  # small radius
+                eps_override = 1e-4
+                if (eps == 0) and (charge != 0):
+                    msg = f"""Warning:
+                        overriding (q={charge}, sig={sig}, eps={eps}) to
+                        ({charge}, {sig_override}, {eps_override}) to avoid singularity"""
+                    if verbose:
+                        print(msg)
+                    sig = sig_override
+                    eps = eps_override
```

Comment:
Is it worth adding an assertion at the end checking that all charged atoms have nonzero $\epsilon$?